src/ticks/expressions.l

```diff
@@ -0,0 +1,78 @@
+%option noyywrap
+
+%{
+#include <stdio.h>
+
+#define YY_DECL int yylex()
+
+#include "exp_engine.h"
+#include "expressions.tab.h"
+
+int lookup_word();
+int lookup_primitive_type();
+
+%}
+
+word [a-zA-Z_][a-zA-Z0-9_]*
+primitive_type "char"|"unsigned char"|"int8_t"|"uint8_t"|"int16_t"|"uint16_t"|"int32_t"|"int"|"uint32_t"|"float"
```

Comment:
+long, long long etc?

Reply:
Resolved by not having a separate token for types at all. Any word is first tried to be a type now (see `lookup_word`)